### ISSUE TITLE
AesCipherDataSource usage

### ISSUE THREAD

**Issue by jovansuzic** [google/ExoPlayer#6472]
Can someone provide us with some code snippet on how to use AesCipherDataSource, to be precise, how to create MediaSource for encrypted video file?
I have found several issues regarding playing encrypted video file and in all of them it is suggested to look at AesCipherDataSource but I couldn't find the way how to create MediaSource using AesCipherDataSource.
As @ojw28 wrote here (https://github.com/google/ExoPlayer/issues/5083), it is suggested to encrypt video using AesCipherDataSink and for decryption to use AesCipherDataSource. 
There is some example with ExtractorMediaSource.Factory but this is now deprecated.

`private void encryptFile(File file, String secretKey) {
    try {
        AssetManager am = this.getAssets();

        // Fully read the input stream.
        InputStream inputStream = am.open("sample_audio.mp3");
        byte[] inputStreamBytes = Util.toByteArray(inputStream);
        inputStream.close();

        // Create a sink that will encrypt and write to file.
        AesCipherDataSink encryptingDataSink = new AesCipherDataSink(
            Util.getUtf8Bytes(secretKey), 
            new DataSink() {
                private FileOutputStream fileOutputStream;
                @Override
                public void open(DataSpec dataSpec) throws IOException {
                    fileOutputStream = new FileOutputStream(file);
                }
                @Override
                public void write(byte[] buffer, int offset, int length) throws IOException {
                    fileOutputStream.write(buffer, offset, length);
                }
                @Override
                public void close() throws IOException {
                    fileOutputStream.close();
                }
            });

        // Push the data through the sink, and close everything.
        encryptingDataSink.open(new DataSpec(Uri.fromFile(file)));
        encryptingDataSink.write(inputStreamBytes, 0, inputStreamBytes.length);
        encryptingDataSink.close();

        Toast.makeText(this, "File encrypted", Toast.LENGTH_SHORT).show();
    } catch (IOException e) {
        e.printStackTrace();
    }
}`

Maybe you can help us @ojw28. 
Thanks in advance.

**Comment by AquilesCanta** [comment]
Does that example work for you? 

Because ExtractorMediaSource is now called ProgressiveMediaSource, but other than that it's exactly the same.

**Comment by jovansuzic** [comment]
Thank you for response.
You were right, it's working with ProgressiveMediaSource.
For anyone else that might encounter the same problem, first check issue (#5083). There you will find example project (ExoPlayerTestProject.zip) posted by @kozmabalazs (https://github.com/google/ExoPlayer/issues/5083#issuecomment-437901562) and also solution provided @ojw28 (https://github.com/google/ExoPlayer/issues/5083#issuecomment-437976602).
To summarize, you don't need to use classes AesDataSource and AesFlushingCipherMod, just CryptedDefaultDataSourceFactory and CryptedDefaultDataSource. In CryptedDefaultDataSource uncomment line 105 that uses AesCipherDataSource and delete line 106. Also, I think the key must be at least 16 bytes. For encryption use method posted above.
